### ISSUE TITLE
Pieces OS, Error! Access denied || My snippets disappeared

### ISSUE THREAD

**Issue by rosie-at-pieces** [pieces-app/support#38]
Original ticket: 

Recent Pieces OS, Error! Access denied

Note: this issue was created automatically from an offline service ticket.

**Comment by rosie-at-pieces** [comment]
![c998a9b004ec-Screenshot_2024_01_14_020126 (1)](https://github.com/pieces-app/support/assets/152189388/af627f14-4a33-460a-8707-f757810ffcba)
![97d168bd213f-Screenshot_2024_01_14_140450 (1)](https://github.com/pieces-app/support/assets/152189388/aee25ad1-2803-4a30-b99e-e54f01e2f58b)

**Comment by rosie-at-pieces** [comment]
We are currently experiencing an issue withn Pieces OS version 7.2.2 where some users are getting "Access denied" messages and are unable to use their Pieces app. We are so sorry for this and are currently working on a solution.

 Please watch this ticket for updates!

**Comment by khedhar108** [comment]
Error is removed but not able to access previously stored snippets.

**Comment by rosie-at-pieces** [comment]
Hi everyone, we have figured out the issue and will be releasing a fix (version 7.2.3) shortly. This was an unfortunate result of One Drive's built in backup mechanism that we don't have control of, but we are now aware of it and will cater for it in the future. 

The new build should prevent this and resolve it for you if you are experiencing it, but if you are still running into the following error then you will likely need some manual intervention.

<img width="445" alt="Screenshot 2024-01-15 at 11 59 01 AM" src="https://github.com/pieces-app/support/assets/152189388/523a651e-e13b-4255-a125-c65d7882a2be">

If you prefer to do this yourself, you can follow the steps in [this video](https://drive.google.com/file/d/1899Q4WK2Gem_5B7E1GhE0AePc5GvQkR2/view?usp=sharing) and **please make sure you remove your com.pieces.os from your OneDrive/Document Folder**. 

Otherwise, please fill out [this form](https://getpieces.typeform.com/to/mCjBSIjF#page=docs-sidebar) and we will hop on a call with you ASAP to get it sorted out.

**Comment by rosie-at-pieces** [comment]
Update - version 7.2.3 is now available! Please see the comment above if you are having further issues.

**Comment by giovanniincammicia** [comment]
Hi, I actually started to have this issue with 7.2.3 (I updated a few minutes ago and restarted the pc).
I'd like to maintain my com.pieces.os folder in OneDrive so that I don't lose my snippets, as far as I understood, they're not saved in cloud (yet? It would be very useful for people working on multiple computers!).

**Comment by rosie-at-pieces** [comment]
@giovanniincammicia we do not recommend maintaining a OneDrive/Documents/Pieces/com.pieces.os directory as it can cause issues with Pieces' own path routing. If you want to backup your snippets we'd recommend saving the com.pieces.os folder within another directory labeled, "Pieces". So, for example, "Documents/Pieces/com.pieces.os". 

You can't keep the com.pieces.os directory on the top of your OneDrive folder because it will conflict with our internal code.

Are you missing your snippets currently or did 7.2.3 resolve that for you?

**Comment by rosie-at-pieces** [comment]
Hi everyone, quick update:

On Thursday, January 11th we released Pieces OS v7.2.0 which included several major and minor changes. Unfortunately, it introduced a problem specifically for Windows users who have OneDrive enabled that caused their snippets to disappear (they aren’t gone, Pieces just wasn’t routing properly to them). 

We released 7.2.2 which subsequently didn’t resolve the issue for every user, and some users started to receive an Access Denied error. 

We released 7.2.3 which again didn’t resolve the issue for every user. 

We are releasing 7.2.4 as soon as we can which we are hoping will resolve everything. This means your snippets should be available again and you should have no connection issues or access denied errors. 

We appreciate your patience as we work out this issue. When it’s available, please update to version 7.2.4 and then do not hesitate to reach out if you are still experiencing issues. We also would appreciate hearing if 7.2.4 worked for you so we can ensure success all around. 

Thanks everyone!!

**Comment by furier** [comment]
Im getting this error with 7.2.4
![image](https://github.com/pieces-app/support/assets/776919/876b0774-08b7-4125-ac29-1e06210a50a8)

**Comment by mark-at-pieces** [comment]
> Im getting this error with 7.2.4
> 
> ![image](https://private-user-images.githubusercontent.com/776919/297191374-876b0774-08b7-4125-ac29-1e06210a50a8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0NDA2MjIsIm5iZiI6MTcwNTQ0MDMyMiwicGF0aCI6Ii83NzY5MTkvMjk3MTkxMzc0LTg3NmIwNzc0LTA4YjctNDEyNS1hYzI5LTFlMDYyMTBhNTBhOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExNlQyMTI1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNjhmN2QxMjAwYjg0YTRiMjMyMDViMjcyOWZkOGRiODY4NzU0MTRlMGM5NmVkZTBhZmE4NTdlZTIwZWFhZTQ4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fY2h4lM--COYxNw23tJEAEilOvBxBjNwSPfgbrWZ5T4)
> 
> as i mentioned in #41 I may have screwd it up even more as I paniced when I first encountered this issue. I first experienced this issue on my home computer after upgrading to 7.2.2, then I used teamviewer to connect to my work pc where I could see that it was currently working and I ziped the `Documents\com.pieces.os` folder and restored it on my home computer before I was aware of the version difference, and since I have tried to restore files from my onedrive trashbin, so maybe I just screwd this up was to much trying to fix it...

hey @furier, we have seen this case and is due to some funky behavior with OneDrive when there is a hybrid of data on the local device and within the cloud. Would you be available for a call? if so please ping me my email is mark@pieces.app, and we can get you up and running

**Comment by furier** [comment]
> > Im getting this error with 7.2.4
> > ![image](https://private-user-images.githubusercontent.com/776919/297191374-876b0774-08b7-4125-ac29-1e06210a50a8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0NDA2MjIsIm5iZiI6MTcwNTQ0MDMyMiwicGF0aCI6Ii83NzY5MTkvMjk3MTkxMzc0LTg3NmIwNzc0LTA4YjctNDEyNS1hYzI5LTFlMDYyMTBhNTBhOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExNlQyMTI1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNjhmN2QxMjAwYjg0YTRiMjMyMDViMjcyOWZkOGRiODY4NzU0MTRlMGM5NmVkZTBhZmE4NTdlZTIwZWFhZTQ4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fY2h4lM--COYxNw23tJEAEilOvBxBjNwSPfgbrWZ5T4)
> > as i mentioned in #41 I may have screwd it up even more as I paniced when I first encountered this issue. I first experienced this issue on my home computer after upgrading to 7.2.2, then I used teamviewer to connect to my work pc where I could see that it was currently working and I ziped the `Documents\com.pieces.os` folder and restored it on my home computer before I was aware of the version difference, and since I have tried to restore files from my onedrive trashbin, so maybe I just screwd this up was to much trying to fix it...
> 
> hey @furier, we have seen this case and is due to some funky behavior with OneDrive when there is a hybrid of data on the local device and within the cloud. Would you be available for a call? if so please ping me my email is [mark@pieces.app](mailto:mark@pieces.app), and we can get you up and running

how do i call you via an email? 😅

**Comment by mark-at-pieces** [comment]
> > > Im getting this error with 7.2.4
> > > ![image](https://private-user-images.githubusercontent.com/776919/297191374-876b0774-08b7-4125-ac29-1e06210a50a8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0NDA2MjIsIm5iZiI6MTcwNTQ0MDMyMiwicGF0aCI6Ii83NzY5MTkvMjk3MTkxMzc0LTg3NmIwNzc0LTA4YjctNDEyNS1hYzI5LTFlMDYyMTBhNTBhOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExNlQyMTI1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNjhmN2QxMjAwYjg0YTRiMjMyMDViMjcyOWZkOGRiODY4NzU0MTRlMGM5NmVkZTBhZmE4NTdlZTIwZWFhZTQ4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fY2h4lM--COYxNw23tJEAEilOvBxBjNwSPfgbrWZ5T4)
> > > as i mentioned in #41 I may have screwd it up even more as I paniced when I first encountered this issue. I first experienced this issue on my home computer after upgrading to 7.2.2, then I used teamviewer to connect to my work pc where I could see that it was currently working and I ziped the `Documents\com.pieces.os` folder and restored it on my home computer before I was aware of the version difference, and since I have tried to restore files from my onedrive trashbin, so maybe I just screwd this up was to much trying to fix it...
> > 
> > 
> > hey @furier, we have seen this case and is due to some funky behavior with OneDrive when there is a hybrid of data on the local device and within the cloud. Would you be available for a call? if so please ping me my email is [mark@pieces.app](mailto:mark@pieces.app), and we can get you up and running
> 
> how do i call you via an email? 😅

I sent you can email, I'll send a link over to you to join :)

**Comment by mark-at-pieces** [comment]
> > > > Im getting this error with 7.2.4
> > > > ![image](https://private-user-images.githubusercontent.com/776919/297191374-876b0774-08b7-4125-ac29-1e06210a50a8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0NDA2MjIsIm5iZiI6MTcwNTQ0MDMyMiwicGF0aCI6Ii83NzY5MTkvMjk3MTkxMzc0LTg3NmIwNzc0LTA4YjctNDEyNS1hYzI5LTFlMDYyMTBhNTBhOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExNlQyMTI1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNjhmN2QxMjAwYjg0YTRiMjMyMDViMjcyOWZkOGRiODY4NzU0MTRlMGM5NmVkZTBhZmE4NTdlZTIwZWFhZTQ4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fY2h4lM--COYxNw23tJEAEilOvBxBjNwSPfgbrWZ5T4)
> > > > as i mentioned in #41 I may have screwd it up even more as I paniced when I first encountered this issue. I first experienced this issue on my home computer after upgrading to 7.2.2, then I used teamviewer to connect to my work pc where I could see that it was currently working and I ziped the `Documents\com.pieces.os` folder and restored it on my home computer before I was aware of the version difference, and since I have tried to restore files from my onedrive trashbin, so maybe I just screwd this up was to much trying to fix it...
> > > 
> > > 
> > > hey @furier, we have seen this case and is due to some funky behavior with OneDrive when there is a hybrid of data on the local device and within the cloud. Would you be available for a call? if so please ping me my email is [mark@pieces.app](mailto:mark@pieces.app), and we can get you up and running
> > 
> > 
> > how do i call you via an email? 😅
> 
> I sent you can email, I'll send a link over to you to join :)

We were able to resolve @furier issue w/ lost snippets, couple different solutions we had to work through.

TLDR: we were able to clean up some folders within the OneDrive Documents location, we found a good backup within the OneDrive recycling bin, and we were able to restore the Pieces folder with inthe proper location in the C:/Users/{USER}/Documents/com.pieces.os location.

Feel free to ping me if anyone is experiencing the same issue or any issues related to this :)

**Comment by HRA42** [comment]
Maybe it would be best to move the data directories over from the `documents` folder into `%appdata%\Mesh Intelligent Technologies, Inc\Pieces OS`

It can happen occasionally that OneDrive pulls from the local documents folder into the `onedrive/documents` folder. Microsoft, themselves, recommends storing user data in the app data directories.

**Comment by nathan-courtney-pieces** [comment]
@HRA42 

We looked into placing our data within AppData, but currently MSIX packages re-route into a containerized structure within C:\Users\<your_user_name>\AppData\Local\Packages\com.MeshIntelligentTechnologi.PiecesOS_84gz00a5z79wr\LocalCache\Local

Which if you ever uninstalled the application, all of your user-generated data would be deleted!

**Comment by HRA42** [comment]
@nathan-courtney-pieces:
That's the detail I missed. What about using a hidden folder inside the user directory?

For example: `\.pieces\Mesh Intelligent Technologies, Inc\Pieces OS`

**Comment by giovanniincammicia** [comment]
> @giovanniincammicia we do not recommend maintaining a OneDrive/Documents/com.pieces.os directory as it can cause issues with Pieces' own path routing. If you want to backup your snippets we'd recommend saving the com.pieces.os folder within another directory labeled, "Pieces". So, for example, "Documents/Pieces/com.pieces.os".
> 
> You can't keep the com.pieces.os directory on the top of your OneDrive folder because it will conflict with our internal code
> 
> Are you missing your snippets currently or did 7.2.3 resolve that for you?

Hi Rosie, thanks, version 7.2.4 solved it. When I first installed Pieces OS, it created the folder Documents/com.pieces.os and OneDrive, as of default, put it in the cloud. Now I have com.pieces.os(deprecated) and a new com.pieces.pfd folder.
I still don't see my snippets, should I do something to reroute Pieces OS? Should I remove the deprecated folder?

**Comment by furier** [comment]
When upgrading to 7.2.5 i now got this error and just when everything was working okay 🤦‍♂️😅
![image](https://github.com/pieces-app/support/assets/776919/8c67181b-d09f-478d-80db-9f3478cd753e)

**Comment by mark-at-pieces** [comment]
> When upgrading to 7.2.5 i now got this error and just when everything was working okay 🤦‍♂️😅 ![image](https://private-user-images.githubusercontent.com/776919/297699302-8c67181b-d09f-478d-80db-9f3478cd753e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1ODgwNzksIm5iZiI6MTcwNTU4Nzc3OSwicGF0aCI6Ii83NzY5MTkvMjk3Njk5MzAyLThjNjcxODFiLWQwOWYtNDc4ZC04MGRiLTlmMzQ3OGNkNzUzZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExOFQxNDIyNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYjA3ZjI0ZTI1MzljNGUzNDVjNmE1Yjk0YTg3M2EyZjcwM2U4YTAzYTE2YmE1NmZkYzU5NDMxNDg2ODZlY2NmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.TW0Lf9kpYKqlqpEeoDHkjLyQmMSeYxABhapKiQubYsk)
> 
> # After rebooting my machine, and Pieces OS restarting and working a lot it seems to be working again...

We have seen this issue, and believe it is related to permission that are not properly refreshed untill after the first install @nathan-courtney-pieces would you mind taking a look into this issue?

**Comment by furier** [comment]
@mark-at-pieces I was to quick to report everything as okay, the desktop app behaves as if it is a fresh install and has backed up the one we fixed previously or so we thought

![image](https://github.com/pieces-app/support/assets/776919/225ddba8-0846-4d1f-99c6-299178483109)

**Comment by furier** [comment]
> > the desktop app behaves as if it is a fresh install.
> > Pieces OS has backed up the one we fixed previously, or so we thought, and created a new one.
> 
> hmmm so after the update you found yourself with your snippets gone?(but the back up is there) @furier
> 
> This is absolutely not correct, going to look into this w/ Nathan.
> 
> I believe this has to do with the fact that you updated, so we will double check into that flow to ensure that this is fully fix.
> 
> Thanks for helping out

snippets gone yes... but in the backup folder ofc, and this time i also have it in a git repository.

**Comment by mark-at-pieces** [comment]
> > > the desktop app behaves as if it is a fresh install.
> > > Pieces OS has backed up the one we fixed previously, or so we thought, and created a new one.
> > 
> > 
> > hmmm so after the update you found yourself with your snippets gone?(but the back up is there) @furier
> > This is absolutely not correct, going to look into this w/ Nathan.
> > I believe this has to do with the fact that you updated, so we will double check into that flow to ensure that this is fully fix.
> > Thanks for helping out
> 
> snippets gone yes... but in the backup folder ofc, and this time i also have it in a git repository.

would you mind sending a picture of your OneDrive/Documents as well?
&& are you on Windows 10 or 11?

**Comment by mark-at-pieces** [comment]
Update:

We found the culprit and believe that it was writing a com.pieces.os from within the suite installer, causing there to be a conflict with your 2 dbs and then it creating the com.pieces.os w/ the timestamp in your C Drive (similar to the Image @furier had above).

We will be launching a new PiecesOS 7.2.6 that will be writing all backups/ and backup dbs into a pieces_backups folder in the C:/Users/{USER}/Documents. (if there are conflicts in the future)

We will be also launching a new version to the Pieces Suite installer.

Thank you again everyone for sticking with us, please reach out and let us know if there are any more failures with this.

**Comment by rosie-at-pieces** [comment]
if you are experiencing any issues with 7.2.6, please follow this link to [schedule a call](https://calendar.google.com/calendar/u/0/appointments/schedules/AcZssZ2iaRWpd1kA2k_xg2KrVi_jiTHo0POiMZSI9WAYsxs5jJ9sU2Rd2zKitLuSFd7l3FhatGzy5Vgt) with @mark-at-pieces.

**Comment by mark-at-pieces** [comment]
Hey everyone, want to thank Adam for allowing me to record some audio in solving a failed Migration status code (22). (That would inhibit Pieces OS from Booting up, displayed on boot)

Here is the link to the Audio ([link](https://drive.google.com/file/d/1JHajvel5Xhob-Yvpw8-9oHCDvGx_aUB9/view?usp=sharing))

Note: More solutions + videos && flow charts to come if you want to take things into your own hands. However, we are also more than happy to set up a person call if you are not feeling comfortable doing this yourself and/or things go wrong.

Disclaimer: If you do decide to resolve this on your own, I would be very careful because it is a possibility, if done incorrectly, that you will lose your snippets.

**Comment by rosie-at-pieces** [comment]
Hey everyone, here is the promised flow chart for recovering your missing snippets and fixing any migration issues by yourself. Two disclaimers:

1. The size of the chart is very daunting, but it's just a step by step to make sure you cover all of your bases. 
2. This is an experimental flow and we can't guarantee that you will have 100% success/won't lose your snippets going through this process. If at any point you need our assistance, please do not hesitate to [schedule a call](https://calendar.google.com/calendar/u/0/appointments/schedules/AcZssZ2iaRWpd1kA2k_xg2KrVi_jiTHo0POiMZSI9WAYsxs5jJ9sU2Rd2zKitLuSFd7l3FhatGzy5Vgt) with us. 

Let us know how it goes!

**Comment by furier** [comment]
> Hey everyone, [here](https://lucid.app/lucidchart/069ed910-f367-45c6-b442-e0b14ad11343/edit?invitationId=inv_f1502037-dac8-44bf-9d7d-fc71d5f5c6fc&page=0_0#) is the promised flow chart for recovering your missing snippets and fixing any migration issues by yourself. Two disclaimers:
> 
> 1. The size of the chart is very daunting, but it's just a step by step to make sure you cover all of your bases.
> 2. This is an experimental flow and we can't guarantee that you will have 100% success/won't lose your snippets going through this process. If at any point you need our assistance, please do not hesitate to [schedule a call](https://calendar.google.com/calendar/u/0/appointments/schedules/AcZssZ2iaRWpd1kA2k_xg2KrVi_jiTHo0POiMZSI9WAYsxs5jJ9sU2Rd2zKitLuSFd7l3FhatGzy5Vgt) with us.
> 
> Let us know how it goes!

it seems to require a login to see the flow chart

**Comment by rosie-at-pieces** [comment]
@furier unfortunately Lucid requires an account even for a public link, but it's free to sign up

**Comment by rosie-at-pieces** [comment]
Resolved, if you are still experiencing issues with this please create a new ticket.
